### PR TITLE
Add additional VCtx functions

### PR DESCRIPTION
**Pull request by RJSonnenberg**
### Description

Add the following functions:
- `VCtx.combine`
- `VCtx.combineAsync`
- `VCtx.bindAndCombineAsync`
- `VCtx.mapAsync`

### CODE REVIEW

File: src/FSharp.Data.Validation/VCtx.fs [hunk 1]
```diff
@@ -45,6 +45,28 @@ module VCtx =
...
+    /// Combines two validation contexts into a single validation context, preserving the result of both contexts.
+    /// </summary>
+    /// <param name="v1">The first validation context.</param>
+    /// <param name="v2">The second validation context.</param>
+    /// <returns>A validation context that combines the results of the two input validation contexts.</returns>
+    /// <remarks>
+    /// This function takes two validation contexts <c>v1</c> and <c>v2</c> and returns a new validation context.
+    /// Prioritizes refuted contexts over disputed contexts and disputed contexts over valid contexts.
+    /// </remarks>
+    let combine (v1: VCtx<'F, 'A>) (v2: VCtx<'F, 'B>): VCtx<'F, 'A * 'B> =
```
Comment (devinlyons) [inline src/FSharp.Data.Validation/VCtx.fs:58]: Is this the same as the `Combine` method in the `validation` CE? Should that method call this function?

**Review by devinlyons — APPROVED** [review]

File: src/FSharp.Data.Validation/VCtx.fs (hunk 1 above)
Comment (RJSonnenberg) [reply src/FSharp.Data.Validation/VCtx.fs:58]: I don't see a `Combine` function anywhere. I see a `MergeSources` in the `VCtxBuilder`. It looks a little different though, with all cases resulting in `RefutedCtx` except for when both are `ValidCtx`. Is that correct? Should I have this `combine` function be identical to `MergeSources` and then have `MergeSources` call this `combine` function?

Comment (RJSonnenberg) [reply src/FSharp.Data.Validation/VCtx.fs:58]: Ok. Is there a reason `MergeSources` makes every result a `RefutedCtx` except when both sources are valid?

Comment (RJSonnenberg) [reply src/FSharp.Data.Validation/VCtx.fs:58]: Per our side conversations, I have update this `combine` function to `mergeSources` to better fit the name in FSharp Computation Expression function definitions. I then updated the `MergeSources` function to use this function. This slightly changed behavior, so I have also updated the unit tests to note this change.

**Pull request merged**